app/models/subject.rb

```diff
@@ -11,7 +13,7 @@
 class Subject < ApplicationRecord
   include ImmutableModel
 
-  NAME_REGEXP = /[a-zA-Z_][\w\.\-]*/
+  NAME_REGEXP = /[a-zA-Z_][\w\.\-]*/.freeze
```

Comment:
Are regexes exempt from frozen string literal?

Reply:
They're immutable so you don't need to freeze them.